### ISSUE TITLE
Add function `forNetwork()` to class `Client`

### ISSUE THREAD

**Issue by deyanzz** [hiero-ledger/hiero-sdk-cpp#201]
### Problem

Currently, it's not possible to execute а request or transaction to а [Hedera Local Node](https://github.com/hashgraph/hedera-local-node). This capability is required for the `C++ SDK` to be able to be run integration tests.
Also, adding this capability will unblock [Issue #164](https://github.com/hashgraph/hedera-sdk-cpp/issues/164).

### Solution

To add new member function to class `Client` - [forNetwork()](https://github.com/hashgraph/hedera-sdk-reference/blob/main/reference/core/Client.md#fornetwork--network-map--string--accountid---client).
Detailed description could be found in [Hedera SDK Reference](https://github.com/hashgraph/hedera-sdk-reference).

### Alternatives

There no other alternatives. Other Hedera's SDKs are using the same approach.

**Comment by deyanzz** [comment]
Moving back to this issue as https://github.com/hashgraph/hedera-sdk-cpp/issues/207 has been completed and merged into `main` branch.

**Comment by deyanzz** [comment]
[Issue #217](https://github.com/hashgraph/hedera-sdk-cpp/issues/217) is now completed and merged into `main` branch, so I'm able to continue my work on this task today.

**Comment by deyanzz** [comment]
This issue is currently blocked by [issue #257](https://github.com/hashgraph/hedera-sdk-cpp/issues/257).

**Comment by deyanzz** [comment]
After [issue #257](https://github.com/hashgraph/hedera-sdk-cpp/issues/257) was merged into main, the `Code Review` for this task could be finished.
